app/models/CategorisationAnswers.scala

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models
+
+import cats.data.{EitherNec, NonEmptyChain}
+import cats.implicits.catsSyntaxTuple2Parallel
+import models.AssessmentAnswer.NoExemption
+import models.ott.{CategorisationInfo, CategoryAssessment}
+import org.apache.pekko.Done
+import pages.{AssessmentPage, HasSupplementaryUnitPage, SupplementaryUnitPage}
+import play.api.libs.json.{Json, OFormat}
+import queries.RecordCategorisationsQuery
+
+final case class CategorisationAnswers(
+  assessmentValues: Seq[AssessmentAnswer],
+  supplementaryUnit: Option[Int]
+)
+
+object CategorisationAnswers {
+
+  implicit lazy val format: OFormat[CategorisationAnswers] = Json.format
+
+  private case class CategorisationDetails(index: Int, assessment: CategoryAssessment, answer: AssessmentAnswer)
+
+  def build(userAnswers: UserAnswers, recordId: String): EitherNec[ValidationError, CategorisationAnswers] =
+    (
+      buildAssessmentDetails(userAnswers, recordId),
+      getSupplementaryUnit(userAnswers, recordId)
+    ).parMapN(CategorisationAnswers.apply)
+
+  private def getSupplementaryUnit(userAnswers: UserAnswers, recordId: String) =
+    userAnswers.getOptionalPageValueForOptionalBooleanPage(
+      userAnswers,
+      HasSupplementaryUnitPage(recordId),
+      SupplementaryUnitPage(recordId)
+    )
+
+  private def buildAssessmentDetails(
+    userAnswers: UserAnswers,
+    recordId: String
+  ): Either[NonEmptyChain[ValidationError], Seq[AssessmentAnswer]] =
+    for {
+      recordCategorisations <- userAnswers.getPageValue(RecordCategorisationsQuery)
+      categorisationInfo    <- getCategorisationInfoForThisRecord(recordCategorisations, recordId)
+      answeredAssessments   <- getAssessmentsFromUserAnswers(categorisationInfo, userAnswers, recordId)
+      _                     <- ensureNoExemptionIsOnlyFinalAnswer(answeredAssessments, recordId)
+      _                     <- ensureHaveAnsweredTheRightAmount(answeredAssessments, categorisationInfo.categoryAssessments.size)
+    //  _                     <- ensureHaveNotSkippedAny(answeredAssessments)
```

Comment:
Need to try and test it manually. Couldn't get this scenario to exist anymore in the automated test, because of how Ewan's stuff writes the answers, I'd just get array index errors if I tried in the test to create answers that "skipped" an assessment.
But I don't trust that isn't fully so want to manually retest and see if the scenario can still happen some other way. Will update this comment when I know :)

Reply:
Ok yeah if you there isn't a way to enter them out of order anymore